ports/espressif/common-hal/mdns/Server.c

```diff
@@ -117,6 +117,11 @@ mp_obj_t common_hal_mdns_server_find(mdns_server_obj_t *self, const char *servic
         next = next->next;
     }
     mp_obj_tuple_t *tuple = MP_OBJ_TO_PTR(mp_obj_new_tuple(num_results, NULL));
+    // The empty tuple object is shared and stored in flash so return early if
```

Comment:
mp_obj_new_tuple already sets the length to n, so instead of special casing num_results == 0, you can remove the assignment of len below.

Reply:
I set it to `added` below in case we can make the wrapper objects for all of the results. So, it may be different than the value at alloc time. Want me to make it compare the current len vs added and then set it if different?

Reply:
urp .. tuples are meant to be static in size, I'm not sure of the impact of truncating one after initial allocation. but the tuple at this point is not seen by python code, so it doesn't matter.